src/rdf4cpp/datatypes/xsd/time/Duration.cpp

```diff
@@ -210,9 +210,46 @@ std::partial_ordering capabilities::Comparable<xsd_duration>::compare(cpp_type c
     }
     return o;
 }
+
+template<>
+nonstd::expected<capabilities::Duration<xsd_duration>::cpp_type, DynamicError>
+capabilities::Duration<xsd_duration>::duration_add(cpp_type const &lhs, cpp_type const &rhs) noexcept {
+    // TODO
```

Comment:
For `xsd:duration` the `Duration` implementation is still missing because I didn't quite know how that type was supposed to be used.

@mcb5637 could you add them?

Reply:
Could this operation instead return `{49, nanos_in_a_month - 50}`? So incrementing/decrementing the months when over/underflow occurs in the nanos?

Reply:
No, because months are not directly convertible to days (Feb is 28/29 days, Jan is 31). So depending on what you do in future with that duration, you need a different result to make it consistent.

Reply:
thats unfortunate. Ok I guess I'll return `Unsupported`